src/interpreter/compiler.ts

```diff
@@ -0,0 +1,290 @@
+    const gotoInstr = constructGotoInstr(PLACEHOLDER_ADDRESS);
+    state.instructions.push(gotoInstr);
+
+    loadFunctionInstr.functionInstrAddress = state.instructions.length;
+
+    // TODO: Replace param declarations when parameter list is supported
+    const functionSymbolTable = addFunctionSymbolTableEntries(
+      [],
+      node,
+      state.symbolTable
+    );
+    if (!isEmptyStatement(node.body)) {
+      node.body.items.forEach((item) => {
+        state.symbolTable = functionSymbolTable;
+        compile(item, state);
+      });
+    }
+    const teardownInstr = constructTeardownInstr();
+    state.instructions.push(teardownInstr);
+
+    gotoInstr.instrAddress = state.instructions.length;
+
+    const assignInstr = constructAssignInstr(
+      getSymbolTableEntry(node.id.name, state.symbolTable)
+    );
+    state.instructions.push(assignInstr);
+  },
+  GotoStatement: (node: GotoStatement, state: CompilerState) => {},
+  Identifier: (node: Identifier, state: CompilerState) => {
+    const loadSymbolInstr = constructLoadSymbolInstr(
+      getSymbolTableEntry(node.name, state.symbolTable)
+    );
+    state.instructions.push(loadSymbolInstr);
+  },
+  IdentifierStatement: (node: IdentifierStatement, state: CompilerState) => {},
+  IfStatement: (node: IfStatement, state: CompilerState) => {},
+  LogicalExpression: (node: LogicalExpression, state: CompilerState) => {
+    let conditionalExpression;
+    if (node.operator === '&&') {
+      conditionalExpression = constructConditionalExpression(
+        node.left,
+        node.right,
+        constructFalseConstant()
+      );
+    }
+    if (node.operator === '||') {
+      conditionalExpression = constructConditionalExpression(
+        node.left,
+        constructTrueConstant(),
+        node.right
+      );
+    }
+    if (conditionalExpression === undefined) {
+      throw new UnsupportedOperatorError(
+        node.operator,
+        UnsupportedOperatorErrorType.LOGICAL
+      );
+    }
+    compile(conditionalExpression, state);
+  },
+  MemberExpression: (node: MemberExpression, state: CompilerState) => {},
+  Program: (node: Program, state: CompilerState) => {
+    const programSymbolTable = addProgramSymbolTableEntries(
+      node,
+      state.symbolTable
+    );
+    const enterProgramInstr = constructEnterProgramInstr(
+      getNumOfEntriesInFrame(programSymbolTable.head)
+    );
+    state.instructions.push(enterProgramInstr);
+    node.body.forEach((item) => {
+      state.symbolTable = programSymbolTable;
+      compile(item, state);
+    });
+  },
+  ReturnStatement: (node: ReturnStatement, state: CompilerState) => {
+    // TODO: Check if return with no argument works correctly.
+    if (isNotUndefined(node.argument)) {
+      if (node.argument.expressions.length > 1) {
+        throw new InvalidCallError('Encountered more than 1 return value');
+      }
+      compile(node.argument.expressions[0], state);
+    }
```

Comment:
Actually, this isn't an error, right? For example, the following program is valid and returns 3:
```c
#include <stdio.h>

int f() {
    return 1, 2, 3;
}

int main() {
    printf("%d\n", f());
}
```